app/routes/upload.step1.tsx

```diff
@@ -0,0 +1,129 @@
+import { ActionFunctionArgs } from '@remix-run/node'
+import { Form, redirect, useNavigation, useRouteError } from '@remix-run/react'
+import { useState } from 'react'
+import { v4 as uuidv4 } from 'uuid'
+import { Button } from '~/components/ui/button'
+import { cn } from '~/lib/utils'
+import { supabase } from '~/supabaseClient'
+
+// Action to handle form submission
+export const action = async ({ request }: ActionFunctionArgs) => {
+  const body = await request.formData()
+  const file = body.get('cv-file') as File | null
+  if (!file) throw Error('Could not get file.')
+  if (
+    file &&
+    ![
+      'application/pdf',
+      'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+    ].includes(file.type)
+  )
+    throw Error('Only PDF and DOCX files are allowed.')
+
+  const uniqueFileName = `${uuidv4()}-${file.name}`
+  const { data, error } = await supabase.storage
+    .from('resumess')
+    .upload(`public/${uniqueFileName}`, file) // this is just for testing now. will be secured later
+
+  if (error) {
+    console.error(error)
+    // @ts-ignore
+    throw Error(error.error)
+  } else {
+    console.log('File uploaded successfully', data)
+  }
+  return redirect('/upload/step2')
+}
+
+export function ErrorBoundary() {
+  // @ts-ignore
+  const { message } = useRouteError()
+  return (
+    <div className="text-red-500 flex flex-col gap-4">
+      <h1>Unexpected Error</h1>
+      <p>{message || 'Something went wrong'}</p>
+      <Button>
+        <a href="/upload/step1">Try again</a>
+      </Button>
+    </div>
+  )
+}
+
+const step1 = () => {
+  const [file, setFile] = useState<File | null>(null)
+  const [error, setError] = useState<string | null>(null)
+
+  const navigation = useNavigation()
+  const isSubmitting = navigation.state === 'submitting'
+
+  const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const uploadedFile = e.target.files?.[0] || null
+
```

Comment:
**suggestion**: Consider extracting the file size limit to a constant for better maintainability and readability.